xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/CommentForm.java

```diff
@@ -58,9 +62,54 @@ public WebElement getContentField()
         return getContainer().findElement(By.tagName("textarea"));
     }
 
+    /**
+     * Find the content field and runs an action using the field.
+     *
+     * @param action The action.
+     */
+    private void runOnContentField(Consumer<WebElement> action)
+    {
+        List<WebElement> textareas = getContainer()
+                                         .findElements(By.tagName("textarea"))
+                                         .stream()
+                                         .filter(it -> !it.getAttribute("class").contains("ckeditor-textarea"))
+                                         .collect(Collectors.toList());
+
+        if (textareas.size() == 0 || !textareas.get(0).isDisplayed()) {
+            WebElement iframe = getContainer().findElement(By.tagName("iframe"));
+            getDriver().switchTo().frame(iframe);
+            action.accept(getDriver().findElement(By.tagName("body")));
+            getDriver().switchTo().parentFrame();
```

Comment:
Maybe we should use the CKEditor page objects here. WDYT?

Reply:
That would make sense indeed.

Reply:
So I just checked and the problem is that the pageobjects module of CKeditor actually depends on test-ui module from platform. So it would introduce a cyclic dependency. So keeping it like that for now.